test/with_dummyserver/test_connectionpool.py

```diff
@@ -313,10 +313,10 @@ def test_socket_options(self):
                 s.close()
 
     def test_disable_default_socket_options(self):
-        """Test that passing None disables all socket options."""
+        """Test that passing empty list disables all socket options."""
         # This test needs to be here in order to be run. socket.create_connection actually tries
         # to connect to the host provided so we need a dummyserver to be running.
-        with HTTPConnectionPool(self.host, self.port, socket_options=None) as pool:
+        with HTTPConnectionPool(self.host, self.port, socket_options=[]) as pool:
```

Comment:
Since it seems we're supporting `None` as well can we instead parametize this test for both `None` and `[]`?

Reply:
Parameterized the test for `None` and `[]`